onager/meta_launcher.py

```diff
@@ -6,6 +6,29 @@
 from .constants import SEP, WSEP, FLAG_ON, FLAG_OFF
 from .history import add_new_history_entry
 
+def filter_cmd_prefix(exclude_variables, cmd_prefix_list, VAR_SEP=' '):
+    exclude_arg_keys = []
+    for key, variable in exclude_variables.items():
+        curr_key_strs = []
+        for v in variable:
+            curr_key_strs.append(key + VAR_SEP + v)
+
+        exclude_arg_keys.append(curr_key_strs)
+
+    filtered_prefix_list = []
+    for cmd_prefix in cmd_prefix_list:
+        all_keys_match = True if exclude_arg_keys else False
+        for curr_key_strs in exclude_arg_keys:
+            all_keys_match &= any(key_str in cmd_prefix for key_str in curr_key_strs)
+
+            if not all_keys_match:
+                break
+
+        if not all_keys_match:
+            filtered_prefix_list.append(cmd_prefix)
+
+    return filtered_prefix_list
```

Comment:
Comments please :)

I'm confused about `all_keys_match`, because it seems like `exclude_arg_keys` could be truthy even if you only have one exclude variable.